core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlPrepareAsyncProcessor.java

```diff
@@ -15,34 +15,78 @@
  */
 package com.datastax.oss.driver.internal.core.cql;
 
+import com.datastax.oss.driver.api.core.cql.ColumnDefinition;
 import com.datastax.oss.driver.api.core.cql.PrepareRequest;
 import com.datastax.oss.driver.api.core.cql.PreparedStatement;
 import com.datastax.oss.driver.api.core.session.Request;
+import com.datastax.oss.driver.api.core.type.UserDefinedType;
 import com.datastax.oss.driver.api.core.type.reflect.GenericType;
+import com.datastax.oss.driver.internal.core.context.DefaultDriverContext;
 import com.datastax.oss.driver.internal.core.context.InternalDriverContext;
+import com.datastax.oss.driver.internal.core.metadata.schema.events.TypeChangeEvent;
 import com.datastax.oss.driver.internal.core.session.DefaultSession;
 import com.datastax.oss.driver.internal.core.session.RequestProcessor;
 import com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures;
+import com.datastax.oss.driver.internal.core.util.concurrent.RunOrSchedule;
 import com.datastax.oss.driver.shaded.guava.common.cache.Cache;
 import com.datastax.oss.driver.shaded.guava.common.cache.CacheBuilder;
+import com.datastax.oss.driver.shaded.guava.common.collect.Streams;
+import com.datastax.oss.protocol.internal.ProtocolConstants;
+import io.netty.util.concurrent.EventExecutor;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.ExecutionException;
 import net.jcip.annotations.ThreadSafe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @ThreadSafe
 public class CqlPrepareAsyncProcessor
     implements RequestProcessor<PrepareRequest, CompletionStage<PreparedStatement>> {
 
+  private static final Logger LOG = LoggerFactory.getLogger(CqlPrepareAsyncProcessor.class);
+
   protected final Cache<PrepareRequest, CompletableFuture<PreparedStatement>> cache;
 
-  public CqlPrepareAsyncProcessor() {
-    this(CacheBuilder.newBuilder().weakValues().build());
+  public CqlPrepareAsyncProcessor(DefaultDriverContext context) {
+    this(CacheBuilder.newBuilder().weakValues().build(), context);
```

Comment:
I understand this is an _internal_ class, but perhaps we should:
1) leave the no-arg constructor and simply add this one that takes a DriverContext
1) call the protected constructor with a `null` for the DriverContext when the no-arg constructor is used
1) Check that the DriverContext is not null in the protected constructor before registering the type change handler
I know that makes it less simple, but it prevents compiler issues in case anyone is doing custom things in their own code with this Processor.

Second comment here. Is it possible we want the DriverContext interface rather than the concrete DefaultDriverContext here? Again, I'm not sure if anyone is digging into the driver to this level, but if `getEventBus()` is defined on the interface, then we can loosen it up and not only allow the DefaultDriverContext.

Reply:
Unfortunately getEventBus() isn't part of DriverContext; that's entirely an implementation-dependent thing.  But I do like the idea of maintaining the no-arg constructor to avoid breaking any existing customizations.  I think I'll fold that in... good suggestion!

Reply:
Constructor re-work can be found in [this commit](https://github.com/datastax/java-driver/pull/1638/commits/ceb2882dea1ac586fb55f8bb0a6cec84681daa85).  Seems to have worked out reasonably cleanly.